src/vm/checker/typecheck/mod.rs

```diff
@@ -173,14 +173,20 @@ impl <'a, 'b> TypeChecker <'a, 'b> {
         let mut local_context = TypingContext::new();
 
         for exp in contract {
-            if type_checker.try_type_check_define(exp, &mut local_context)?
-                .is_none() {
-                    // was _not_ a define statement, so handle like a normal statement.
-                    type_checker.type_check(exp, &local_context)?;
+
+            let mut result = type_checker.try_type_check_define(exp, &mut local_context);
+            if let Err(ref mut error) = result {
+                if !error.has_expression() {
+                    error.set_expression(exp);
                 }
+            }
+            let result = result?;
```

Comment:
Not a bug, but I personally dislike reusing variable names like this.  I would use a different variable name for the `let mut result = ...` line (e.g. `result_res`) to hint to the reader that this variable is a `Result<...>`.

Reply:
Addressed with https://github.com/blockstack/blockstack-core/pull/1041/commits/f22ddc80c685a28da08e76f57d85986695db107d

Reply:
Thanks!